Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +121,39 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+// https://w3c.github.io/trusted-types/dist/spec/#get-trusted-type-compliant-string-algorithm
+String getTrustedTypeCompliantString(const TrustedType expectedType, JSGlobalObject& global, JSValue input, const String& sink, ShouldConvertNullToEmptyString shouldConvertNullToEmptyString)
+{
+    VM& vm = global.vm();
+
+    if (expectedType == TrustedType::TrustedHTML) {
+        if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, input))
+            return trustedHTML->toString();
+    } else if (expectedType == TrustedType::TrustedScript) {
+        if (auto* trustedScript = JSTrustedScript::toWrapped(vm, input))
+            return trustedScript->toString();
+    } else if (expectedType == TrustedType::TrustedScriptURL) {
+        if (auto* trustedScriptURL = JSTrustedScriptURL::toWrapped(vm, input))
+            return trustedScriptURL->toString();
+    } else {
+        ASSERT_NOT_REACHED();
+        return nullString();
+    }
+    RefPtr scriptExecutionContext = jsDynamicCast<JSDOMGlobalObject*>(&global)->scriptExecutionContext();
```

Comment:
We might need to check for scriptExecutionContext.

Reply:
Not sure what the best thing to do would be in the case where it doesn't exist? Currently I've got an ASSERT_NOT_REACHED() which doesn't neccessarily feel right?